File: ex1_add1/src/SimpleClosure.scala
object SimpleClosure {
    def main(args: Array[String]): Unit = {
        println(add(5)(4))

        val add10 = add(10)
        println(add10(90))
        println(add10(999))
    }

    def add(x: Int): Int => Int = {
        (y: Int) => x + y
    }
}

File: ex1_add1/src/SimpleClosure.cs
using System;

public class SimpleClosure
{
    private static Func<int, int> Add(int x)
    {
        return y => x + y;
    }

    public static void Main()
    {
        Console.WriteLine(Add(5)(4));

        Func<int, int> add10 = Add(10);

        Console.WriteLine(add10(90));
        Console.WriteLine(add10(999));
    }
}
